pymongo/read_preferences.py

```diff
@@ -143,6 +149,11 @@ def document(self) -> dict[str, Any]:
         if self.__max_staleness != -1:
             doc["maxStalenessSeconds"] = self.__max_staleness
         if self.__hedge not in (None, {}):
+            warnings.warn(
```

Comment:
The `document` helper should not raise a warning. We already raised a warning in the constructor. We do need to keep the warning in `hedge()` since we will remove that api in 5.0.